woocommerce/woocommerce-gateway-stripe
# Update the UPE Gateway title in the checkout depending the enabled methods

# Changes proposed in this Pull Request:

Issue: #1919
Fixes #1919

This PR changes the UPE gateway title in the checkout depending on the enabled payment methods.
If there is more than one (1) payment method enabled the title is: `Popular payment methods`, but if there is only one, then that payment method title is used (ex: `Pay with giropay`)

__NOTE__: This solution does not use the gateway `$this->title` nor `$this->description`, instead it adds a new `$this->checkout_title` with the value `Popular payment methods` (like in WC Payments), so the merchant to go back to the normal Stripe gateway.

**BEFORE:**
<img width="630" alt="Screen Shot 2021-09-21 at 13 57 07" src="https://user-images.githubusercontent.com/407542/134225447-f8c90dc0-1d61-4f55-a523-e53bcb6e5db1.png">

**AFTER:**
<img width="335" alt="Screen Shot 2021-09-22 at 14 05 54" src="https://user-images.githubusercontent.com/407542/134389563-a2c0955c-d3e1-4e34-91ee-f1d96ab81bae.png"><img width="340" alt="Screen Shot 2021-09-22 at 14 06 54" src="https://user-images.githubusercontent.com/407542/134389595-055572a3-20a6-4734-945e-d72ae25cb19f.png">


# Testing instructions

1. Make sure UPE is enabled (`Try the new payment experience (Early access)` is checked)
2. Enable at least 2 payment methods
3. Go to the Shop and add at least one product to the cart
4. Go to the Shortcode Checkout
5. Check that the Gateway title is `Popular payment methods` (like in the screenshot above)
6. Also check that the description `Pay with your credit card via Stripe` is not present below the tile; it should only say: `Test mode...`
7. 4. Go to the Blocks Checkout
8. Check that the Gateway title is `Popular payment methods`
9. Go to the Stripe payment settings, and enable only the `giropay` payment method (disable the rest)
10. Go to the Shortcode Checkout
11. Check that the Gateway title is `Pay with giropay` (like in the screenshot above)
12. Go to the Blocks Checkout
13. Check that the Gateway title is `Pay with giropay` (like in the screenshot above)
